Command/MigrateConfigCommand.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+namespace Lexik\Bundle\JWTAuthenticationBundle\Command;
+
+use Jose\Bundle\JoseFramework\JoseFrameworkBundle;
+use Jose\Component\Checker\ClaimCheckerManager;
+use Jose\Component\Core\JWK;
+use Jose\Component\Core\JWKSet;
+use Jose\Component\Encryption\JWEBuilder;
+use Jose\Component\Encryption\JWELoader;
+use Jose\Component\KeyManagement\JWKFactory;
+use Lexik\Bundle\JWTAuthenticationBundle\Services\KeyLoader\KeyLoaderInterface;
+use Symfony\Bundle\FrameworkBundle\Command\AbstractConfigCommand;
+use Symfony\Component\Config\Definition\Processor;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\DependencyInjection\Compiler\ValidateEnvPlaceholdersPass;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Extension\ConfigurationExtensionInterface;
+use Symfony\Component\DependencyInjection\Extension\ExtensionInterface;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * @author Nicolas Cabot <n.cabot@lexik.fr>
+ *
+ * @final
+ */
+#[AsCommand(name: 'lexik:jwt:migrate-config', description: 'Migrate LexikJWTAuthenticationBundle configuration to the Web-Token one.')]
+class MigrateConfigCommand extends AbstractConfigCommand
+{
+    /**
+     * @deprecated
+     */
+    protected static $defaultName = 'lexik:jwt:migrate-config';
+
+    /**
+     * @var KeyLoaderInterface
+     */
+    private $keyLoader;
+
+    /**
+     * @var string
+     */
+    private $signatureAlgorithm;
+
+    /**
+     * @var string
+     */
+    private $passphrase;
+
+    public function __construct(
+        KeyLoaderInterface $keyLoader,
+        string $passphrase,
+        string $signatureAlgorithm
+    ) {
+        parent::__construct();
+        $this->keyLoader = $keyLoader;
+        $this->passphrase = $passphrase === '' ? null : $passphrase;
+        $this->signatureAlgorithm = $signatureAlgorithm;
+    }
+
+    public function isEnabled(): bool
+    {
+        $requirements = [
+            JoseFrameworkBundle::class => 'web-token/jwt-bundle',
+            JWKFactory::class => 'web-token/jwt-key-mgmt',
+            ClaimCheckerManager::class => 'web-token/jwt-checker',
+            JWEBuilder::class => 'web-token/jwt-encryption',
+            JWELoader::class => 'web-token/jwt-signature',
+        ];
+
+        foreach (array_keys($requirements) as $requirement) {
+            if (!class_exists($requirement)) {
+                return false;
+            }
+        }
+
+        return true;
```

Comment:
I suggest to rather throw a meaningful exception in `execute()` telling the developer to install the required packages if they are not available, so they can catch the issue when running this command (as they will probably discover the integration in the docs). Btw I'm pretty sure this command should never end up being registered in production, right? Can we remove its service definition in non debug mode maybe?

Reply:
The service definition is now removed unless the debug mode in enabled